### DISCUSSION TITLE
Is there any way to get the response from ExecuteScriptAsync synchronously.

### DISCUSSION THREAD

**Discussion by JitinKansal** [MicrosoftEdge/WebView2Feedback#3935] [Q&A]
Problem:- we are able to call the JS function at server end from our application and also we are getting the response/return value in Asynchronously But we need the value synchronously. So, Is there any way to do that?

I am using webview2 control and using this code for calling the function of JS at server end.

```
Dim json As New Web.Script.Serialization.JavaScriptSerializer
json.MaxJsonLength = Int32.MaxValue
Dim jsoncontent = json.Serialize(New Object() {"S"})
Await WViform_browser.ExecuteScriptAsync($"JavaScriptFucntionAtServerEnd({jsoncontent})")
```

And this is the JS code at server end

```js
window.page = {
JavaScriptFucntionAtServerEnd:()=> { JavaScriptFucntionAtServerEnd();},
}
function JavaScriptFucntionAtServerEnd(){
window.chrome.webview.postMessage(true);
}
```

**Comment by JitinKansal** [comment]
Please suggest the solution for this issue it's urgent.

**Comment by champnic** [comment]
Hey @JitinKansal - thanks for the question.

Which is the part that you need returned synchronously? Do you need ExecuteScriptAsync to be sync, or is the await ok there?

In general sync is going to be difficult to achieve because the WebView2 is out-of-process. Turning it sync would involve pumping messages until you get a return value. We definitely recommend using async programming techniques if possible.
